### PR TITLE
Remove obsolete TypeSymbolWithAnnotations.Create method

### PR DESCRIPTION
**Pull request by cston**
`TypeSymbolWithAnnotations.Create(referenceType)` now creates an instance with `IsNullable==null`.

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler please review.

File: src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs [hunk 1]
```diff
@@ -138,31 +138,14 @@ internal TypeSymbolWithAnnotations SubstituteType(TypeSymbolWithAnnotations prev
...
-            return SubstituteType(previous, withTupleUnification: true);
-        }
-
         internal TypeSymbolWithAnnotations SubstituteType(TypeSymbol previous, bool withTupleUnification)
         {
             var result = SubstituteType(previous);
 
-            if (withTupleUnification)
+            // Make it a tuple if it became compatible with one.
+            if (withTupleUnification && !previous.IsTupleCompatible())
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs]: >&& !previous.IsTupleCompatible() [](start = 37, length = 32)

Is condition ```!previous.IsTupleCompatible()``` redundant? Will it be checked by ```TransformToTupleIfCompatible``` anyway?

File: src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs [hunk 2]
```diff
@@ -317,7 +317,8 @@ private void CreateEmbeddedAttributeItselfIfNeeded(DiagnosticBag diagnostics)
             var boolArray = TypeSymbolWithAnnotations.Create(
                 ArrayTypeSymbol.CreateSZArray(
                     boolType.ContainingAssembly,
-                    TypeSymbolWithAnnotations.Create(boolType)));
+                    TypeSymbolWithAnnotations.Create(boolType)),
+                isNullableIfReferenceType: null);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs]: >isNullableIfReferenceType: null [](start = 16, length = 31)

Does this have to be specified explicitly?

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceDestructorSymbol.cs [hunk 3]
```diff
@@ -65,7 +65,7 @@ protected override void MethodChecks(DiagnosticBag diagnostics)
         {
             var syntax = GetSyntax();
             var bodyBinder = this.DeclaringCompilation.GetBinderFactory(syntaxReferenceOpt.SyntaxTree).GetBinder(syntax, syntax, this);
-            _lazyReturnType = TypeSymbolWithAnnotations.Create(bodyBinder.GetSpecialType(SpecialType.System_Void, diagnostics, syntax));
+            _lazyReturnType = TypeSymbolWithAnnotations.Create(nonNullTypesContext: this, bodyBinder.GetSpecialType(SpecialType.System_Void, diagnostics, syntax));
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceDestructorSymbol.cs]: >TypeSymbolWithAnnotations.Create(nonNullTypesContext: this, bodyBinder.GetSpecialType(SpecialType.System_Void, diagnostics, syntax)); [](start = 30, length = 133)

I think we can safely say that it is not annotated (not nullable)

**Comment by 333fred** [comment]
@cston looks like there's a compile error

**Pull request merged**